### PR TITLE
[url_launcher] Remove deprecations

### PR DESCRIPTION
**Pull request by Zazo032**
## Description

Removed Android deprecation warnings by adding `@SuppressWarnings` to deprecated method, and `@RequireApi` to the non-deprecated method targeting N (API 24), when it was introduced.

Android Studio weirdly suggests to remove this new redundant expression, but not sure why:
image

## Related Issues

Fixes https://github.com/flutter/flutter/issues/37616

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/url_launcher/url_launcher/android/src/main/java/io/flutter/plugins/urllauncher/WebViewActivity.java [hunk 1]
```diff
@@ -89,7 +92,9 @@ public void onCreate(Bundle savedInstanceState) {
     final Map<String, String> headersMap = new HashMap<>();
     for (String key : headersBundle.keySet()) {
       final String value = headersBundle.getString(key);
-      headersMap.put(key, value);
+      if (value != null) {
+        headersMap.put(key, value);
+      }
```
Comment (Zazo032) [inline packages/url_launcher/url_launcher/android/src/main/java/io/flutter/plugins/urllauncher/WebViewActivity.java:97]: Android Studio suggested this change as `value` may not be initialized.

**Comment by shinriyo** [comment]
@Zazo032 
`Conflicting files`

**Comment by Zazo032** [comment]
Fixed by  @mvanbeusekom on #3008

**Comment by shinriyo** [comment]
finally?

**Comment by shinriyo** [comment]
`url_launcher 5.5.3` is included it?

**Pull request closed without merging**